docs/learn/expert-guides/deploy-up-with-lsp23.md

```diff
@@ -147,94 +151,87 @@ When deploying your Universal Profile, you might want to initialize it with some
-  const universalReceiverPermissionsValue =
-    '0x0000000000000000000000000000000000000000000000000000000000060080'; // REENTRANCY & SUPER_SETDATA permissions
-}
-```
-
-Now we will want after deployment to have a main controller that has all the permissions on the Universal Profile. We will start by adding this address under our previous constants:
-
-```typescript
-const mainController = '0x3303Ce3b8644D566271DD2Eb54292d32F1458968'; // this is just a random address
-```
-
-Then we will create the data key and data value for the permissions of the main controller:
-
-```typescript
-async function main() {
-  // previous code
-
-  const mainControllerPermissionsKey =
-    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
-    MAIN_CONTROLLER.slice(2);
-  // main controller will have all permissions on the Universal Profile
-  const mainControllerPermissionsValue = ALL_PERMISSIONS;
-}
-```
-
-#### Create the `AddressPermissions[]` Array length data key and data value
-
-```typescript
-async function main() {
-  // previous code
-
-  const addressPermissionsLengthDataKey =
-    ERC725YDataKeys.LSP6['AddressPermissions[]'].length;
-
-  const abiCoder = new AbiCoder(); // instantiate the abiCoder
-  const addressPermissionsLengthDataValue = abiCoder.encode(['uint256'], [2]); // Address Permissions array length
-}
-```
-
-#### Create the AddressPermissions[] array data keys and data values
-
-```typescript
-async function main() {
-  // previous code
-
-  const addressPermissionsArrayFirstElementKey =
-    ERC725YDataKeys.LSP6['AddressPermissions[]'].index +
-    abiCoder.encode(['uint128'], [0]).slice(34); // remove the 0x and the first 16 bytes of the index
-  console.log(addressPermissionsArrayFirstElementKey);
-  const addressPermissionsArrayFirstElementValue = UNIVERSAL_RECEIVER_ADDRESS; // first element is the Universal Receiver
-
-  const addressPermissionsArraySecondElementKey =
-    ERC725YDataKeys.LSP6['AddressPermissions[]'].index +
-    abiCoder.encode(['uint128'], [1]).slice(34); // remove the 0x and the first 16 bytes of the index
-  const addressPermissionsArraySecondElementValue = mainController; // second element is the main controller
+  // create the permissions data keys
+  const setDataKeysAndValues = erc725.encodeData([
+    { keyName: 'LSP3Profile', value: lsp3DataValue }, // LSP3Metadata data key and value
+    {
+      keyName: 'LSP1UniversalReceiverDelegate',
+      value: UNIVERSAL_RECEIVER_ADDRESS,
+    }, // Universal Receiver data key and value
+    {
+      keyName: 'AddressPermissions:Permissions:<address>',
+      dynamicKeyParts: [UNIVERSAL_RECEIVER_ADDRESS],
+      value: erc725.encodePermissions({
+        REENTRANCY: true,
+        SUPER_SETDATA: true,
+      }),
+    }, // Universal Receiver Delegate permissions data key and value
+    {
+      keyName: 'AddressPermissions:Permissions:<address>',
+      dynamicKeyParts: [MAIN_CONTROLLER],
+      value: erc725.encodePermissions({
```

Comment:
Maybe for the main controller permissions we could re-use the constant from the lsp-smart-contracts? But keeping it like this is also very good IMO and very clear. So I would even argue and leave it as it is.

Maybe for the main controller permissions we could re-use the constant from the lsp-smart-contracts? But keeping it like this is also very good

Reply:
hmm i am not sure i like the fact that they can choose which permisison they want to give at deployment

Reply:
OK we agreed to keep it like this as it makes it crystal clear 👌